### PR TITLE
Fix stream synchronization in MTMG graph construction

### LINKED ISSUE #4236: MTMG Use of streams
Review MTMG use of streams.  We allocate a stream pool for MTMG to allow for different host threads to copy data to the same GPU in parallel.  We have a couple of different ways we access streams which sometimes make it easy to grab the wrong stream.

### PR DESCRIPTION
**Pull request by ChuckHastings**
Restructure to pass stream instead of handle and synchronize appropriately.

Closes #4236

### CODE REVIEW

File: cpp/include/cugraph/mtmg/detail/per_device_edgelist.hpp [hunk 1]
```diff
@@ -62,13 +62,13 @@ class per_device_edgelist_t {
...
    *
-   * @param handle             MTMG resource handle - used to identify GPU resources
+   * @param stream_view        CUDA stream view
    * @param device_buffer_size Number of edges to store in each device buffer
    * @param use_weight         Whether or not the edgelist will have weights
    * @param use_edge_id        Whether or not the edgelist will have edge ids
    * @param use_edge_type      Whether or not the edgelist will have edge types
    */
-  per_device_edgelist_t(cugraph::mtmg::handle_t const& handle,
+  per_device_edgelist_t(rmm::cuda_stream_view stream_view,
```
Comment (seunghwak) [inline cpp/include/cugraph/mtmg/detail/per_device_edgelist.hpp:71]: Something tedious, but I guess our convention is to pass the handle as the first input argument and stream_view as the last input argument. Should we better keep this convention?

**Review by seunghwak — APPROVED** [review]
LGTM

File: cpp/include/cugraph/mtmg/detail/per_device_edgelist.hpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/include/cugraph/mtmg/detail/per_device_edgelist.hpp:71]: Yes, I should have caught that.  I will make that change.

**Pull request merged**